vochain/start.go

```diff
@@ -335,3 +349,23 @@ func AminoKeys(key crypto25519.PrivKey) (private, public []byte, err error) {
 
 	return aminoKey, aminoPubKey, nil
 }
+
+// restoreStateFromSnapshot overrides the node data with a given snapshot
+func restoreStateFromSnapshot(dataDir, src string) error {
+	log.Infof("deleting old datadir ... %s", dataDir)
+	if _, err := os.Stat(dataDir); err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("cannot check if destination directory exists")
+	} else if err != nil {
+		if err := os.RemoveAll(dataDir); err != nil {
+			return fmt.Errorf("cannot clean directory before applying the snapshot")
+		}
+	}
```

Comment:
i don't think this previous checks and nested ifs are necessary.  `os.RemoveAll()` returns success if the path didn't exist in the first place.
(also, the `else if err != nil` looks wrong?)

> RemoveAll removes path and any children it contains. It removes everything it can but returns the first error it encounters. If the path does not exist, RemoveAll returns nil (no error). If there is an error, it will be of type *PathError.

anyway i'd replace those 8 lines with these 4
```
	log.Infof("cleaning up old datadir %s before applying the snapshot", dataDir)
	if err := os.RemoveAll(dataDir); err != nil {
		return fmt.Errorf("cannot clean old datadir: %w", err)
	}
```